File: HW1/src/ack.py
#Maitri Mangal
#CS471 Assignment 1

#! /usr/bin/env python
import sys

def ack(m, n):
	if m == 0:
		return n+1
	elif n == 0 and m > 0:
		return ack(m-1, 1)
	else:	
		return ack(m-1, ack(m, n-1))

if len(sys.argv) != 3:
  print("%s usage: [NUMBER] [NUMBER]" % sys.argv[0])
  exit()

print("ack() : "+ str( ack(int(sys.argv[1]), int(sys.argv[2]))))

File: HW1/ack.rb
#! /usr/bin/env ruby

def ack(m, n)
	return n+1 if m == 0
	return ack(m-1, 1) if n == 0 and m > 0
	return ack(m-1, ack(m, n-1)) if n > 0 and m > 0
	return -1
end

if ARGV.length != 2 
  puts "ack.rb usage: [NUMBER] [NUMBER]" 
  exit
end

puts "ack() : " + ack(ARGV[0].to_i, ARGV[1].to_i).to_s
